packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import uniqueid from 'lodash.uniqueid';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+    * Whether or not the Switch is enabled ("ON").
+    */
+  isOn: PropTypes.bool,
+
+  /**
+    * If true, the user won't be able to toggle the switch.
+    */
+  isDisabled: PropTypes.bool,
+
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+
+  /**
+    * Callback fired when the state is changed.
+    */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isOn: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
+  isOn,
+  isDisabled,
+  onChange,
+  labelText,
+  ...customProps
+}) => {
+  const [focused, setFocus] = useState(false);
+  const sliderButtonRef = useRef(null);
+
+  const handleClick = (event) => {
+    setFocus(true);
+    sliderButtonRef.current.focus();
+    onChange(event);
+  };
+
+  const handleOnKeyDown = (event) => {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      event.preventDefault();
+      setFocus(true);
+    }
+  };
+
+  const handleOnBlur = () => {
+    setFocus(false);
+  };
+
+  const handleKeyUp = (event) => {
+    if (event.keyCode === KeyCode.KEY_TAB) {
+      setFocus(true);
+    } else if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      setFocus(true);
+      onChange(event);
+    }
+  };
+
+  const switchClassNames = cx([
+    'switch',
+    customProps.className,
+  ]);
+
+  const trayClassNames = cx([
+    'tray',
+    { selected: isOn },
+    { unselected: !isOn },
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const sliderClassNames = cx([
+    'slider',
+    { 'is-disabled': isDisabled },
+    { 'is-focused': focused },
+  ]);
+
+  const statusLabelText = isOn ? SWITCH_STATE.ON : SWITCH_STATE.OFF;
+
+  const attributes = { ...customProps };
+  const switchButtonId = (attributes.id) ? attributes.id : `terra-switch-button-${uniqueid()}`;
```

Comment:
I wonder if we could make this a first class prop for accessibility purposes instead and not generate one internally ourselves 🤔

Reply:
It was First class prop but then we might need to have it as required field too. since we will be needing ID to reference switch button. I thought of generating it on it's own.  
User can still pass ID as [custom props](https://github.com/cerner/terra-core/blob/1c002b1da07d995fc126d14539617437b6371dd7/packages/terra-switch/src/Switch.jsx#L100) to switch button which will override the internal ID when provided.

Reply:
The field components (SelectField, InputField, TextareaField, etc..) require an ID. I think it makes sense to require an ID for this component. Currently, this generates a new ID every render. This will make testing complicated and provide unnecessary dom updates. If you open the demo page with the inspector and toggle the component you'll see the component updates every toggle change and blur/focus changes.

Reply:
Makes sense Added Id as Prop to switch component : [here](https://github.com/cerner/terra-core/blob/895c4fad7bb180e27e1919296ac90eeaa805dea1/packages/terra-switch/src/Switch.jsx#L19-L23)